### PR TITLE
new(docs): add an auto-generated eBPF timeline

### PR DESCRIPTION
**Pull request by Andreagit97**
The idea behind this PR is to add a timeline for all ebpf features, taking advantage of the scraping logic already present in this repo. The timeline uses the information collected in `data/feature-versions.yaml`(the same file used for generating tags in the documentation).

I see at least 2 possible use cases here:
* a new kernel version is out and you want to check in a unique place all the new features introduced
* you need to develop something with some kernel version constraints and you can easily see what  you can you and what you cannot

New Timeline:

Screenshot from 2024-10-19 12-45-41

### CODE REVIEW

File: tools/feature-gen/timeline.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		}
+
+		// In our case it will never happen to have 2 tags with the same Major and Minor, but in any case we return true if they are equal
+		return v1Minor <= v2Minor
+	})
+
+	for _, tag := range tags {
+		fmt.Fprintf(&sb, "\n## :octicons-tag-24: %s\n\n", tag)
+		for _, feature := range timeline[tag] {
+			fmt.Fprintf(&sb, "* **%s** [%s](https://github.com/torvalds/linux/commit/%s) (%s)\n", feature.Name, feature.Commit[:7], feature.Commit, feature.Type)
```
Comment (dylandreimerink) [inline tools/feature-gen/timeline.go:77]: The spellcheck is triggering on the short hand terms used in the feature names. I suggest changing these to inline code blocks which are ignored by the spellchecker 

```suggestion
			fmt.Fprintf(&sb, "* `%s` [%s](https://github.com/torvalds/linux/commit/%s) (%s)\n", feature.Name, feature.Commit[:7], feature.Commit, feature.Type)
```

**Review by dylandreimerink — APPROVED** [review]
This is great! Never thought of using the existing data this way. Thank you

File: tools/feature-gen/timeline.go (hunk 1 above)
Comment (Andreagit97) [reply tools/feature-gen/timeline.go]: ty!

Comment (Andreagit97) [reply tools/feature-gen/timeline.go]: ahahhaah fair enough! ty

**Pull request merged**